src/components/StudyRoom/StudyRoomHome.js

```diff
@@ -10,26 +10,20 @@ import Typography from "@mui/material/Typography";
 import Divider from "@mui/material/Divider";
 import {useNavigate} from "react-router";
 import axios from "axios";
-
-
+import {GetAuthentication} from "../Authentication/Authentification";
 
 export default function StudyRoomHome() {
     const navigate = useNavigate();
     const [roomData, setRoomData] = React.useState([]);
-    const email = JSON.parse(localStorage.getItem("email"));
+    const user = GetAuthentication();
 
     useEffect(() => {
-        //user needs to be logged in to access
-        if (email === undefined || email === '') {
-            navigate("/login");
-        }
         getData();
-    }, [email])
+    })
```

Comment:
Did this compile with no dependency array?

Reply:
yes